models/contact.js

```diff
@@ -1,9 +1,47 @@
+const {Schema, model} = require('mongoose')
 const Joi = require("joi");
+const {handleMongooseError} = require('../helpers')
+// const Schema = mongoose.Schema;
 
-const addShema = Joi.object({
+const contactSchema = new Schema(
+  {
+    name: {
+      type: String,
+      required: [true, 'Set name for contact'],
+    },
+    email: {
+      type: String,
+    },
+    phone: {
+      type: String,
+    },
+    favorite: {
+      type: Boolean,
+      default: false,
+    },
+  }
+)
+contactSchema.post('save',handleMongooseError )
+
+const addSchema = Joi.object({
   name: Joi.string().required(),
   email: Joi.string().required(),
```

Comment:
you should validate email and phone properly

Reply:
Can I do it in next  hw(04-auth)?

const  phoneno = /^\+?([0-9]{2})\)?[-. ]?([0-9]{4})[-. ]?([0-9]{4})$/;
const addSchema = Joi.object({

  email: Joi.string().email().required(),
  phone: Joi.string().pattern(phoneno).required(),

});